tests/default_host_resolver_test.c

```diff
@@ -143,7 +143,7 @@ static int s_test_default_with_ipv6_lookup_fn(struct aws_allocator *allocator, v
     aws_string_destroy((void *)host_name);
     aws_host_resolver_release(resolver);
     aws_event_loop_group_release(el_group);
-    ASSERT_SUCCESS(aws_global_thread_creator_shutdown_wait_for(10));
+    aws_thread_join_all_managed();
```

Comment:
This should be accomplished by aws_io_library_clean_up() calling aws_common_library_clean_up(), it should not be a requirement for downstream users to manage. Lifetime should be clear: init -> run application -> clean_up.